internal/controller/component_build_controller_test.go

```diff
@@ -397,6 +397,82 @@ var _ = Describe("Component build controller", func() {
 				return isCreatePaCPullRequestInvoked
 			}, timeout, interval).Should(BeTrue())
 		})
+
+		It("should manage pull secrets for nudging", func() {
+			nudgedComponentKey := types.NamespacedName{Namespace: namespace, Name: "nudged-component"}
+			nudgedComponentSAKey := getComponentServiceAccountKey(nudgedComponentKey)
+
+			createComponent(nudgedComponentKey)
+			nudgedComponentImageRepo := createImageRepositoryForComponent(nudgedComponentKey)
+			defer deleteImageRepository(nudgedComponentKey)
+
+			// Simulate Image Controller and add push secret of nudged Component to its Service Account
+			nudgedComponentSA := waitServiceAccount(nudgedComponentSAKey)
+			nudgedComponentSA.Secrets = append(nudgedComponentSA.Secrets, corev1.ObjectReference{Name: nudgedComponentImageRepo.Status.Credentials.PushSecretName})
+			Expect(k8sClient.Update(ctx, &nudgedComponentSA)).To(Succeed())
+
+			component1SAKey := getComponentServiceAccountKey(component1Key)
+			createCustomComponentWithoutBuildRequest(componentConfig{
+				componentKey: component1Key,
+				finalizers:   []string{PaCProvisionFinalizer}, // Simulate PaC provision done
+			})
+			component1ImageRepo := createImageRepositoryForComponent(component1Key)
+			defer deleteImageRepository(component1Key)
+
+			// Create a component with nudge reference
+			component2SAKey := getComponentServiceAccountKey(component2Key)
+			component2 := getSampleComponentData(component2Key)
+			component2.Finalizers = append(component2.Finalizers, PaCProvisionFinalizer) // Simulate PaC provision done
+			component2.Spec.BuildNudgesRef = []string{
+				nudgedComponentKey.Name,
+			}
+			createComponentCustom(component2)
+			component2ImageRepo := createImageRepositoryForComponent(component2Key)
+			defer deleteImageRepository(component2Key)
+			// Check that the pull secret of nudging Component was linked to the nudged Component build pipeline Service Account
+			waitServiceAccountLinkedSecret(nudgedComponentSAKey, component2ImageRepo.Status.Credentials.PullSecretName, false)
+			// Check that a pull secret was not linked from a not nudging Component
+			waitServiceAccountLinkedSecretGone(nudgedComponentSAKey, component1ImageRepo.Status.Credentials.PullSecretName, false)
+			// Also ensure, that the secret wasn't linked to a wrong Service Account
+			waitServiceAccountLinkedSecretGone(component1SAKey, component2ImageRepo.Status.Credentials.PullSecretName, false)
+			waitServiceAccountLinkedSecretGone(component2SAKey, component2ImageRepo.Status.Credentials.PullSecretName, false)
```

Comment:
I don't understand this line, are we checking that the pull secret for Component 2's ImageRepository is _not_ linked to Components 2's Service Account? It seems like that _should_ be linked, no?

Reply:
push secret should but not pull secret

Reply:
The pull secret should be linked to nudged Component, but not for nudging Component (component2).

Reply:
Ah right, pull vs push secret :+1: